C/Cpp_include/c4ConnectedClient.hh

```diff
@@ -23,29 +22,40 @@ struct C4ConnectedClient  : public fleece::RefCounted,
                             C4Base
 {
     /// Creates a new ConnectedClient
+    /// \note It will automatically starts the client, no need to call `start()`.
+    ///
+    /// @param params  Connected Client parameters.
+    /// @result A new \ref C4ConnectedClient, or NULL on failure.
     static Retained<C4ConnectedClient> newClient(const C4ConnectedClientParameters &params);
                     
     /// Gets the current revision of a document from the server.
-    virtual litecore::actor::Async<C4DocResponse> getDoc(slice,
-                                                         slice,
-                                                         slice,
-                                                         bool) noexcept=0;
+    /// You can set the `unlessRevID` parameter to avoid getting a redundant copy of a
+    /// revision you already have.
+    /// @param docID  The document ID.
+    /// @param collectionID  The name of the document's collection, or `nullslice` for default.
+    /// @param unlessRevID  If non-null, and equal to the current server-side revision ID,
+    ///                   the server will return error {WebSocketDomain, 304}.
+    /// @param asFleece  If true, the response's `body` field is Fleece; if false, it's JSON.
+    /// @result An async value that, when resolved, contains either a `C4DocResponse` struct
+    ///          or a C4Error.
+    virtual litecore::actor::Async<C4DocResponse> getDoc(slice docID,
+                                                         slice collectionID,
+                                                         slice unlessRevID,
+                                                         bool asFleece) noexcept=0;
```

Comment:
Remove `noexcept` here and below. These methods can throw exceptions, because they call ConnectedClient methods not marked `noexcept`. (The public C++ API does use exceptions to signal errors.)